### ISSUE TITLE
Illegal offset type in class-ludicrousdb.php on line 1477

### ISSUE THREAD

**Issue by mcorkum** [stuttter/ludicrousdb#78]
Refers to this line: 
`$this->dbhname_heartbeats[$dbh]['last_used'] = microtime( true );`

Something here is throwing tons warnings per page load. Version 4.2.0 (the actual tag) doesn't throw this warning, something seems to have changed since then. 

Any ideas?

**Comment by magix01** [comment]
I also have the same problem, using php 7.2, MySQL 8, WP version latest (5.2.2) throws alot of warnings and i am not sure if i can use it in production before it gets resolved. Trying to use standard master/slave config.

**Comment by thefrosty** [comment]
To be clear. This error is shown when on the latest master branch. Not the current tag.

**Comment by musthavetickets** [comment]
I'm getting this after using latest branch. I have mariadb 10.4, php 7.3 and latest WP. It's on line 
 PHP Warning:  Illegal offset type in /var/xxxxx/plugins/ludicrousdb/ludicrousdb/includes/class-ludicrousdb.php on line 1482" while reading response header from upstream.

The line is:
$this->dbhname_heartbeats[$dbh]['last_used'] = microtime( true );

**Comment by romainmenke** [comment]
This should be fixed with : #86

**Comment by JJJ** [comment]
> Maybe the problem is MySQL 8 since it wont allow saving 0000 date when making a post?

WordPress internally overrides that, along with other incompatible MySQL modes, so it should not be a problem here:

```
		'NO_ZERO_DATE',
		'ONLY_FULL_GROUP_BY',
		'STRICT_TRANS_TABLES',
		'STRICT_ALL_TABLES',
		'TRADITIONAL',
```

This issue is resolved, and other ongoing work to improve this is happening in #88.
